fix(mediarequest): refactor to .save() instead of .update() for TypeORM

gauthier-th opened the pull request:
#### Description

The .update() method of TypeORM doesn't come with all the functionalities of the .save() method. Its goal is to partially update a Repository without doing any extra operation. This introduces issues with event from the database subscriber not triggered correctly, because of this partial update: https://github.com/typeorm/typeorm/issues/2809#issuecomment-451914877
A positive side effect of this change is that web push are now fixed.

This PR rollback to .save() instead of .update() to avoid the aforementioned issues, ~~at the cost of some lifehook issues. These issues are happening with PostgreSQL because the .save() method shouldn't be used inside an Entity Listener, an Event Subscriber should be used instead: https://orkhan.gitbook.io/typeorm/docs/listeners-and-subscribers#what-is-an-entity-listener~~

EDIT (By @fallenbagel):

This PR consolidates all database‐accessing lifecycle hooks into the MediaRequestSubscriber as this is the recommended way of doing database calls.
See: https://orkhan.gitbook.io/typeorm/docs/listeners-and-subscribers#what-is-an-entity-listener.

It also tightens the cascade options on the `MediaRequest→Media` relation to only `insert` and `remove` to avoid `hook‐within‐hook` recursions that could cause unexpected behavior.

These changes should guard against recursive hook invocations in any database backend and simplify the entity lifecycle.

Upstream PR also opened: https://github.com/sct/overseerr/pull/4124

#### Screenshot (if UI-related)

#### To-Dos

- [ ] Successful build `pnpm build`
- [ ] Translation keys `pnpm i18n:extract`
- [ ] Database migration (if required)

gauthier-th committed:
fix(mediarequest): refactor to .save() instead of .update() for TypeORM

The .update() method of TypeORM doesn't come with all the functionalities of the .save()
method. Its goal is to partially update a Repository without doing any extra operation. This
introduces issues with event from the database subscriber not triggered correctly, because of
this partial update: https://github.com/typeorm/typeorm/issues/2809#issuecomment-451914877

This PR rollback to .save() instead of .update() to avoid the aforementioned issues, at the
cost of some lifehook issues. These issues are happening with PostgreSQL because the .save()
method shouldn't be used inside an Entity Listener, an Event Subscriber should be used instead:
https://orkhan.gitbook.io/typeorm/docs/listeners-and-subscribers#what-is-an-entity-listener

gauthier-th committed:
fix(mediarequest): move methods modifying MediaRequest to its Subscriber

fallenbagel commented:
The last commit reverts this revert but its just to show what works for postgres to @gauthier-th (can be reverted if there is a way to fix this). It seems `.save()` breaks it completely for postgres where it goes on an infinite loop of hitting the `afterUpdate` unfortunately. 

From my testing, the .update changes I brought do not affect any subscribers that need an Entity as they're added for request declines which afaik is not needed by any subscribers. 

However, there is currently a weird ui issue that I have not delved deeply into. Even after an approval/rejection (though the status changes from `pending` to `requested` in the status badge, the request is still approvable/reject-able so you have to press twice for it to disappear which ofcourse tries to add the same media to services again).

fallenbagel committed:
fix(mediasubscriber): use event.manager for parent media updates on remove

Replace `getRepository(Media)` calls with `event.manager` in the `afterRemove` hook so that
parent-media status resets run within the same transaction/QueryRunner (important for postgresql.
Doesnt affect sqlite).

fallenbagel committed:
fix(mediarequest entity): narrow cascade to insert & remove to prevent hook recursion

Restrict cascade options on the MediaRequest→Media relation to only `insert` and `remove`to avoid
nested subscriber/AfterUpdate recursion when saving entities.

fallenbagel commented:
Superseded by #1631
